### PR TITLE
Add method to get initialize url.

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Adds get_initialize_url method to app interface API
* Has optional argument for next url
* Returns the URL relative to the host (protocol, host name and port must be supplied)

### Reviewer guidance
Can you generate the URL?

Note that I did not add the protocol hostname, and port, because the port can be dynamically changed by the `--port` argument, so it is best left to the invoker.

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by indirectlylit** [comment]
what functionality is this in support of, and how should I test it?

**Comment by rtibbles** [comment]
Basically, it's just a single place for the Android (or any other app) wrapper to get the URL that it should initially navigate to in order to load Kolibri.

The URL is optional, in case it has saved state of the last URL visited and wants to restore that state after a restart.

**Comment by rtibbles** [comment]
To test it - I would just check that giving it a URL like `/ar/learn/#/channels/` gives a URL that when you load it first goes to initialize and then redirects you to the URL you supplied.

File: kolibri/plugins/app/utils.py [hunk 1]
```diff
@@ -21,6 +23,19 @@ def register(self, **kwargs):
...
                 self._capabilities[capability] = kwargs[capability]
 
+    def get_initialize_url(self, next_url=None):
+        if not self.enabled:
+            raise RuntimeError("App plugin is not enabled")
+        # Import here to prevent a circular import
+        from kolibri.core.device.models import DeviceAppKey
+
+        url = reverse(
+            "kolibri:kolibri.plugins.app:initialize", args=(DeviceAppKey.get_app_key(),)
```
Comment (kollivier) [inline kolibri/plugins/app/utils.py:33]: `get_app_key()` returns a UUID type, but `reverse` is expecting a string, so we need to convert when passing as an arg. I tried to push a fix for this myself but got a repo authentication error.

**Review by kollivier — CHANGES_REQUESTED** [review]
PR works great with the aforementioned conversion, once that is in I'll approve this for merging, thanks! :)

File: kolibri/plugins/app/utils.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/plugins/app/utils.py:33]: Oh - that's odd. It should really always return a string. Will look into it.

**Comment by rtibbles** [comment]
Got a weird test environment error, so have rebased too for good measure.

**Review by kollivier — APPROVED** [review]
LGTM!

**Pull request merged**